### ISSUE TITLE
Other Level of Description

### ISSUE THREAD

**Issue by kfherbert** [harvard-library/aspace-import-excel#16]
We are using your plugin to import a large amount (16,000 records) of data from an old records management system, and use the  "Other Level" for Description for most items as they were described at the box level. The plug in allows us to input "Other Level" in the Description Level column, without issue the "Other Level" field is blank. But without a column to input the text for the name of that level, when ever we need to edit a record we need to add it.  And as the column is blank I can not use batch find replace to add the information.

I noticed that you have this as a question on the spreadsheet:
"Should there be a column for Other Level of Description in Basic Information? And/or can it default to "unspecified"?"

I wanted to know if there was a plan to add this to the plug in.  
--

**Comment by bobbi-SMR** [comment]
Hi; sorry I've been slammed at work.  I hope to take a look at this today, and see if I can get you the answer you want.

--Bobbi

**Comment by bobbi-SMR** [comment]
OK,  I've taken a look; it wasn't clear from your email exactly what you were doing to get the results you were getting.  Let me make sure we are on the same page:

- Using the template spreadsheet, you've added "Other Level" to the Data Validation List for the "Description Level" column
- You then are successful in specifying archival objects with a "Description Level" of "Other Level"
- Once you get those objects loaded, they are all missing the free text "Other Level" field that, were you to create the archival object manually, is a required field.

Assuming this is correct (and I tried it myself):

Because the initial version of this plugin was to satisfy the needs of Harvard's archivists, and they don't use "other level" very much, we just didn't include it, or its testing.

Because the free-text "Other Level" field is required in ArchiveSpace when the level "Other Level" is selected, I would have expected that, when my plugin tried to save the archival object, the ArchivesSpace validation would catch it, and reject that line.

I've proven to my satisfaction that this doesn't happen (I'll probably file an issue, or otherwise let them know that this is a problem for Aspace).

I am going to have to modify the template to allow for "Other Level" to be selected, and add an "other level" column (defaulting to "unspecified").  When I've made the changes, tested them, and modified the User Documentation, I'll announce it to the aspace users list, as well as let you know.

Hopefully, this won't take more than a day or two (she wrote optimistically).
